st/clitests/awsiam_spec.py

```diff
@@ -191,7 +201,148 @@ def policy_tests():
+
+	#list user attached policies
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser").execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 2):
+        print('List Policies Test failed')
+        quit()
+    result.command_response_should_have("iampolicy")
+
+    #list user attached policies with path prefix filter
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser", \
+        path_prefix="/listpolicy/").execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 1):
+        print('List Policies Test failed')
+        quit()
+    result.command_response_should_have("iampolicy2")
+
+	#list user policies for user not having any policies attached
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser2") \
+        .execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 0):
+        print('List Policies Test failed')
+        quit()
+
+    #list user policies for wrong username
+    AwsIamTest('List User Attached Policies').list_attached_user_policies("asdhj") \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+	#detach-user-policy
+    AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn).execute_test() \
+        .command_is_successful()
+    AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn2).execute_test() \
+        .command_is_successful()
+
+    #detach-user-policy for non existing user
+    result = AwsIamTest('Detach User Policy').detach_user_policy("hkhk", arn) \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+    #detach-user-policy with non existing policy
+    result = AwsIamTest('Detach User Policy').detach_user_policy("testUser", "arn:seagate:iam::103947660857:policy/oiuou") \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+    #detach-user-policy with non attached policy
+    result = AwsIamTest('Detach User Policy').detach_user_policy("testUser", arn) \
+        .execute_test(negative_case=True).command_should_fail().command_error_should_have("NoSuchEntity")
+
+	#list user policies after detaching the policies
+    result = AwsIamTest('List User Attached Policies').list_attached_user_policies("testUser").execute_test()
+    total_policies = get_attached_user_policy_list_count(result.status.stdout, "ATTACHEDPOLICIES")
+    if(total_policies != 0):
+        print('List Policies Test failed')
+        quit()
+
+    #delete-policy
+    result = AwsIamTest('Delete Policy').delete_policy(arn).execute_test().command_is_successful()
```

Comment:
Handle below negative scenario-
DeletePolicy before detaching.

Reply:
This scenario is not handled. Right now the policy is getting deleted even if it is attached to any user. So we need to add that validation in delete policy then this ST can be added.

Reply:
Please do the needful. Its important

Reply:
That will be taken care in another Ticket which I have already listed in the TODO list and informed to Basav.

Reply:
ok plz keep a track. Resolving conversation hence